inspect_return_value.py

```diff
@@ -0,0 +1,170 @@
+"""
+<Program Name>
+  inspect_return_value.py
+
+<Author>
+  Sachit Malik <i.sachitmalik@gmail.com>
+
+<Started>
+  June 6, 2017
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Inspections constitute an important part of in-toto. In this script,
+  we inspect the return value of a step. The user supplies an integer
+  to the program(as compared to which the user wants to test various
+  boolean operators), along with path to the corresponding link file
+  (which link file he wants to test in), and also a bash style boolean
+  operator.
+
+  Suppose the link file is located at /user/abc/def/package.45gh325.link
+  and the user wants to check whether for the corresponding step(hence the link file),
+  the corresponding return field contains an integer less than 5.
+
+  The usage would be as follows:
+  python inspect_return_value.py -l /user/abc/def/package.45gh325.link -o lt 5
+
+  Please note: the various bash style boolean commands are listed as follows-
+  * bash-style-boolean-operator
+    --eq ... equal
+    --ne ... not equal
+    --lt ... less than
+    --le ... less than or equal
+    --gt ... greater than
+    --ge ... greater than or equal
+
+  General usage:
+  python inspect_return_value.py -l <path/to/the/link/file> -o <operator> <integer>
+
+"""
+import os
+import sys
+import argparse
+import in_toto.log
+from in_toto.models.link import Link as link_import
+import securesystemslib.exceptions
+
+
+def inspect_return_value(link, operator, integer):
+    """
+    <Purpose>
+      A function which performs the inspection as described above depending on various arguments.
+
+    <Arguments>
+      link:
+        the path to the link file
+
+      operator:
+        bash style boolean operator
+
+      integer:
+        the integer to which the return value should be compared
+        Note: the comparison happens treating the integer as a consequent,
+        for example: given "lt" as an operator, it will inspect whether
+        (the return value) <= (the given integer)
+
+    <Exceptions>
+      None
+
+    <Returns>
+      Boolean
+
+    """
+
+    imported_link = link_import.read_from_file(link)
+    if operator == 'eq':
+      if integer == imported_link.return_value:
+        return True
+
+    elif operator == 'ne':
+      if integer != imported_link.return_value:
+        return True
+
+    elif operator == 'lt':
+      if integer > imported_link.return_value:
+        return True
+
+    elif operator == 'le':
+      if integer >= imported_link.return_value:
+        return True
+
+    elif operator == 'gt':
+      if integer < imported_link.return_value:
+        return True
+
+    elif operator == 'ge':
+      if integer <= imported_link.return_value:
+        return True
+
```

Comment:
I think, since we are using the choices in argparse, it won't reach to a stage that some other operator other than the available choices is passed to this function. Since, on doing that, while parsing, argparse will itself raise an exception. So this stage when an operator other than the ones in the choices list is entered, won't ever be reached. What do you think?